include/pybind11/cast.h

```diff
@@ -139,6 +153,95 @@ inline PyThreadState *get_thread_state_unchecked() {
 #endif
 }
 
+PYBIND11_NOINLINE inline bool try_implicit_conversion(const bool already_loaded[], const std::vector<PyObject *> &py_args) {
+    auto &internals = get_internals();
+    auto &instances = internals.implicit_instances_stack.top().first;
+    auto &arg_types = *internals.implicit_instances_stack.top().second;
+    auto &imp_conv = internals.implicit_conversions_cpp;
+
+    // Store the constructors we find; we only actually construct if we find a constructor for all
+    // unloaded types:
+    std::vector<std::pair<size_t, void *(*)(void *)>> constructors;
+    for (size_t i = 0; i < arg_types.size(); i++) {
+        if (already_loaded[i]) continue;
+        auto type = Py_TYPE(py_args[i]);
+
+        for (auto &conversion : imp_conv[arg_types[i]]) {
+            if (PyType_IsSubtype(type, conversion.first)) {
+                auto &constructor = conversion.second;
+                if (constructor) {
+                    constructors.emplace_back(i, constructor);
+                }
+                // Allow a null constructor reference to abort conversion attempts
+                break;
+            }
+        }
+        // If the i'th argument didn't load normally, and we didn't find a constructor, we can't do implicit conversion
+        if (constructors.empty() || constructors.back().first != i)
```

Comment:
Simpler logic: add ``bool found=false;`` above for loop over ``imp_conv``, set to true if found,  and check for it here.